### ISSUE TITLE
Expose clip_grad_norm_ through the browsergrad-jit torch alias

### ISSUE THREAD

**Issue by vijayksingh** [unlocalhosted/browsergrad#9]
## Gap

Current `browsergrad-grad` source exposes `torch.nn.utils.clip_grad_norm_`, but the published `@unlocalhosted/browsergrad-jit` 0.8.2 torch alias has no `torch.nn.utils` clipping surface. Browser course kernels using the JIT runtime therefore cannot exercise the standard PyTorch training-loop boundary without a compatibility layer or a NumPy imitation.

## Requested surface

```python
total_norm = torch.nn.utils.clip_grad_norm_(
    parameters,
    max_norm,
    norm_type=2.0,
    error_if_nonfinite=True,
    foreach=None,
)
```

Important parity points:

- one total norm over all present gradients, viewed as one vector
- ignore parameters whose `.grad is None`
- return the pre-clip total norm as a Tensor
- modify present gradients in-place when possible
- preserve dtype/device grouping and promotion semantics
- support `error_if_nonfinite`
- state sparse, foreach, accelerator, and mixed-device limits explicitly
- register `torch.nn.utils` and `torch.nn.utils.clip_grad_norm_` in the alias

## Useful regression fixture

```python
weight = torch.nn.Parameter(torch.tensor([1.0, -1.0], dtype=torch.float32))
bias = torch.nn.Parameter(torch.tensor([0.5], dtype=torch.float32))
weight.grad = torch.tensor([3.0, 4.0])
bias.grad = torch.tensor([-12.0])

before = torch.nn.utils.clip_grad_norm_([weight, bias], 5.0, error_if_nonfinite=True)
```

`before` should be scalar Tensor `13`; both gradients should receive the same epsilon-aware coefficient; parameter values must remain unchanged. Cases should also cover `grad=None`, no present gradients, infinity norm, non-finite failure, and the exact-threshold epsilon edge.

**Comment by vijayksingh** [comment]
> *This was generated by AI during triage.*

## Agent Brief

**Category:** enhancement
**Summary:** Expose PyTorch-compatible gradient-norm clipping through `torch.nn.utils.clip_grad_norm_`.

**Current behavior:**
The JIT torch alias has no `torch.nn.utils` namespace and no `clip_grad_norm_`, so standard training loops must use BrowserGrad-specific workarounds even though parameter gradients and in-place gradient replacement are available.

**Desired behavior:**
Provide `torch.nn.utils.clip_grad_norm_(parameters, max_norm, norm_type=2.0, error_if_nonfinite=False, foreach=None)`. It should compute the total norm across present gradients, return the pre-clipping norm as a scalar Tensor, and scale all present gradients by one shared coefficient when clipping is required. Parameters with `grad is None` are ignored. Unsupported execution options must fail explicitly.

**Key interfaces:**
- `torch.nn.utils` — public compatibility namespace.
- `clip_grad_norm_()` — iterable handling, norm calculation, scalar Tensor return, shared in-place gradient scaling, and non-finite policy.
- Parameter gradients — preserve shape, dtype, session, and parameter values while updating only gradients.

**Acceptance criteria:**
- [ ] Empty iterables and parameters with only `None` gradients return a scalar zero Tensor without modifying parameters.
- [ ] L2 clipping returns the pre-clip total norm and leaves below-threshold gradients unchanged.
- [ ] Above-threshold gradients are all scaled by the same coefficient and their combined norm respects `max_norm`.
- [ ] Infinity norm is supported; other promised norm types match PyTorch within supported dtype tolerance.
- [ ] `error_if_nonfinite=True` rejects NaN/Inf total norms; false follows the documented compatibility behavior.
- [ ] Unsupported `foreach`, sparse, mixed-session, or device cases fail clearly rather than silently changing semantics.

**Out of scope:**
- Optimizer stepping or automatic clipping inside optimizers.
- Unsupported accelerator/sparse semantics beyond explicit diagnostics.

**Comment by vijayksingh** [comment]
> *This was generated by AI during triage.*

## Verified outcome

**State:** wontfix — already implemented

Implemented in https://github.com/unlocalhosted/browsergrad/commit/4d7181a9e14ba8c69e8714277da4172b361afcf4: torch.nn.utils.clip_grad_norm_ is registered through both BrowserGrad and torch aliases, returns the pre-clip scalar Tensor, applies shared p-norm/infinity-norm scaling, ignores None gradients, and rejects unsupported foreach/device/mixed-session cases explicitly. Fresh verification passed JIT build, both typechecks, lint, 27 unit tests, 345 integration tests, architecture checks, packed-release tests, and the required JIT semantic-permute WebGPU gate.

Closing as completed and replacing the ready state with `wontfix` per the repository triage state machine.
